trailblazer/cli/core.py

```diff
@@ -89,11 +95,14 @@ def init(context, reset, force):
 
 
 @base.command()
-@click.pass_context
-def scan(context):
+@inject
+def scan(
+    job_service: JobService = Provide[Container.job_service],
```

Comment:
Vert cool stuff. I'm not sure when this was merged but I'm lacking doc strings and type hints in `trailblazer/containers.py` and `trailblazer/server/wiring.py`. It's not super intuitive, and without some supporting information it risks becoming black magic only you can deal with 😅

Reply:
I don't thunk we should make ourselves dependent on this package without properly discussing it first. The dependency injector is not properly maintained any longer and we run the risk of having to re do this in the future. See here the initial developers stopped maintaining it and there is no one taking it over right now...
https://github.com/ets-labs/python-dependency-injector/pull/765 

Not sure if we should make ourselves vulnerable like this @henrikstranneheim, @islean , @diitaz93  what do you think?

Currently I think we solve dependency injection via de config.

Reply:
Yeah, it will need to be replaced 👍

Reply:
@seallard @ChrOertlin I don't really see the benefit from merging this unapproved option if we know that we do not want it. I would have preferred it if we left out the injection until we have agreed on an option in situations like this.

Reply:
The library was not introduced in this PR but I agree. Sometimes it is also nice to try things out, we can use trailblazer as a demo for whatever we settle on. There are 200 other repos to apply it in 😅

Reply:
I don't think the solution we have (with the config) is working great. Usually when I comment on dependencies being instantiated in a CLI command or endpoint, it turns out it is impossible to do it in the config. So we need a better solution

Reply:
Oh, fair enough, if it was not introduced here you can disregard 😄 Did not intend for it to sound harsh, but if it was introduced here, it would probably have been nice to use a maintained package immediately. I agree that the config solution does have some issues though.

Reply:
I really like the syntax of this package but it is not viable long term :/ While it has many users, there is another package that actually is maintained and is more widely used

Reply:
@seallard @islean @ChrOertlin Since the dependency is not new to TB, I see harm in using it. But, for the aforementioned 200 other apps it would be great with a discussion and well maintained packages before embarking on that odessey.